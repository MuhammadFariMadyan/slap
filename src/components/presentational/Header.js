import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { MdPlayArrow, MdClose } from "react-icons/md";
import "typeface-roboto";

import Button from "./Button";

const Container = styled.div`
  width: 100%;
  height: 64px;
  background: white;
  display: flex;
  align-items: center;
  padding: 24px;
  box-sizing: border-box;
`;

const Name = styled.input`
  border: 0px;
  border-bottom: 1px solid white;
  font-family: Roboto;
  font-size: 18px;
  font-weight: bold;
  color: #333;
  outline: none;
  padding: 8px 0px;

  &:focus {
    border-color: #ccc;
  }
`;

const IconPlay = styled(MdPlayArrow)`
  font-size: 20px;
  color: white;
  margin-right: 4px;
`;

const IconClose = IconPlay.withComponent(MdClose);

export const PlayButton = styled(Button)`
  background: #4834d4;
  margin-right: 4px;
  margin-left: auto;

  &:hover {
    background: #3a2aaf;
  }
`;

export const CloseButton = styled(Button)`
  background: #eb4d4b;

  &:hover {
    background: #cc4341;
  }
`;

const Header = ({
  className,
  name,
  onClickPlay,
  onClickClose,
  onChangeName
}) => (
  <Container className={className}>
    <Name onChange={onChangeName} value={name} />
    <PlayButton icon={<IconPlay />} caption="Play" onClick={onClickPlay} />
    <CloseButton icon={<IconClose />} caption="Close" onClick={onClickClose} />
  </Container>
);

Header.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string,
  onClickClose: PropTypes.func,
  onClickPlay: PropTypes.func,
  onChangeName: PropTypes.func
};

export default Header;
