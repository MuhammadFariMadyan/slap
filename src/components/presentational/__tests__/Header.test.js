import React from "react";
import { shallow } from "enzyme";

import Header, { PlayButton, CloseButton } from "../Header";

describe("Header", () => {
  const setup = propOverrides => {
    const props = {
      name: "My Project",
      onClickPlay: jest.fn(),
      onClickClose: jest.fn(),
      onChangeName: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<Header {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("should calls 'onClickPlay' callback while play clicked", () => {
    const { wrapper, props } = setup();

    wrapper.find(PlayButton).simulate("click");

    expect(props.onClickPlay).toBeCalled();
  });

  it("should calls 'onClickClose' callback while close clicked", () => {
    const { wrapper, props } = setup();

    wrapper.find(CloseButton).simulate("click");

    expect(props.onClickClose).toBeCalled();
  });
});
