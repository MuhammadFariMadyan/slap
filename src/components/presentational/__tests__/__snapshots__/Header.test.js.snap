// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should renders default 1`] = `
<styled.div>
  <styled.input
    onChange={[Function]}
    value="My Project"
  />
  <Styled(Button)
    caption="Play"
    icon={<Styled(Component) />}
    onClick={[Function]}
  />
  <Styled(Button)
    caption="Close"
    icon={<Styled(Component) />}
    onClick={[Function]}
  />
</styled.div>
`;
